cache/expirationcache/expiration_cache.go

```diff
@@ -122,15 +148,23 @@ func (e *ExpiringLRUCache[T]) Put(key string, val *T, ttl time.Duration) {
 		val:            val,
 		expiresEpochMs: expiresEpochMs,
 	})
+
+	if e.onAfterPut != nil {
+		e.onAfterPut(e.lru.Len())
+	}
 }
 
 func (e *ExpiringLRUCache[T]) Get(key string) (val *T, ttl time.Duration) {
 	el, found := e.lru.Get(key)
 
 	if found {
+		e.onCacheHit(key)
```

Comment:
onCacheHit/onCacheMiss and preExpirationFn can't be nil, it will be initialized in `NewCacheWithOnExpired`. onAfterPut will not be initialized and can be nil, because it calls len() and it uses lock inside -> can be avoided.